src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -122,4 +127,68 @@ public static Tag parseTag(String tag) throws ParseException {
         }
         return tagSet;
     }
+
+    /**
+     * Parses a {@code String date} into a {@code LocalDate}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code date} is invalid.
+     */
+    public static LocalDate parseTime(String time) throws ParseException {
+        requireNonNull(time);
+        String trimmedTime = time.trim();
+        try {
+            return LocalDate.parse(trimmedTime, Lesson.FORMATTER);
+        } catch (DateTimeParseException e) {
+            throw new ParseException(Lesson.DATE_MESSAGE_CONSTRAINTS);
+        }
+    }
```

Comment:
Since the function is `parseTime`, maybe should change `Lesson.DATE_MESSAGE_CONSTRAINTS` to `Lesson.TIME_MESSAGE_CONSTRAINTS`? Just a suggestion haha

Reply:
Agreed, I was halfway through implementing when I realized the "time" parameter should've been named "date" :) I will change "time" to "date" later.